### DISCUSSION TITLE
How to configure pm2

### DISCUSSION THREAD

**Discussion by WesleyjanLacerda** [Ylianst/MeshCentral#5757] [Show and tell]
The purpose of the post is just informative for those who want to use pm2, as I searched a little here and didn't find this configuration, follow the step by step below:

OS: Ubuntu 20.04
Version: 1.1.11
Node: v16.20.2

With the application already configured, run:
`sudo npm install -g pm2`
Create a file `pm2.config.js` in the system root, to have the directory like this:
`/home/ubuntu/meshcentral/pm2.config.js`

Inside the pm2.config.js file you can create other configurations, but the basics for how it works are this:
```
module.exports = {
    apps: [
      {
        name: 'meshcentral',
        script: 'node_modules/meshcentral',
        autorestart: true,
        watch: false,
        max_memory_restart: '1G',
        error_file: 'logs/error.log',
        out_file: 'logs/out.log',
        log_file: 'logs/combined.log',
        time: true
      }
    ]
  };
  
```
After creating and saving the file, run the command `sudo pm2 start pm2.config.js`.

With the steps above followed, the application will be running automatically by pm2, check with ` sudo pm2 ls`, but details in the documentation <https://pm2.keymetrics.io/docs/usage/pm2-doc-single-page/>.

![image](https://github.com/Ylianst/MeshCentral/assets/74305459/57f017c7-3b28-43d7-922b-143b613a4d34)

**Comment by si458** [comment]
oh wow thanks!
if you wanted to, you could do a PR and add this to the docs?
add a section to the full install guide - https://github.com/Ylianst/MeshCentral/blob/master/docs/docs/install/install2.md

**Comment by SomeGuru** [comment]
Nice job @WesleyjanLacerda ,

I use PM2 on some other applications and this makes it easier to deploy services in that realm... Thanks for the information...

SomeGuru
